### PR TITLE
WIP: Allow specifying which components of q to output on each fgout grid

### PR DESCRIPTION
**Pull request by rjleveque**
This requires https://github.com/clawpack/geoclaw/pull/624 

The main `Makefile.dclaw` now uses `fgout_module.f90` from `geoclaw` rather than a separate one for D-Claw.

(I also added a couple directories `tests/geoclaw_radial_slide` and `tests/geoclaw_bouss_radial_slide` for testing geoclaw SWE and Bouss on this radial problem with water on the hill instead of dirt, perhaps useful to others, but these don't have fgout grids.)

I updated the example in `examples/radial_slide` to add
```
fgout.q_out_vars = [1,4,8]
```
to output h, hm, and eta at 100 fgout frames.

There's also a new `pyvista_plotting.py` module to make a 3D plot using PyVista, with the surface colored by mass fraction (hm / h).

A few frames shown below:
PyVistaFrame0001
PyVistaFrame0033
PyVistaFrame0078

### CODE REVIEW

File: src/2d/dig/Makefile.dclaw [hunk 1]
```diff
@@ -23,14 +23,14 @@ COMMON_MODULES += \
...
- $(GEOLIB)/adjointsup_module.f90
+ $(GEOLIB)/adjointsup_module.f90 \
+ $(GEOLIB)/fgout_module.f90 \
 
 #list of common modules needed for dclaw codes
 COMMON_MODULES += \
  $(DIGLIB)/digclaw_module.f90 \
  $(DIGLIB)/qinit_module.f90 \
  $(DIGLIB)/auxinit_module.f90 \
- $(DIGLIB)/fgout_module.f90 \
```
Comment (kbarnhart) [inline src/2d/dig/Makefile.dclaw:33]: @rjleveque , @dlgeorge I suggest we delete dclaw/src/2d/dig/fgout_module.f90.

We only have it it as a placeholder that is no longer needed b/c Randy updated the GEOLIB/fgout_module.f90. 

Thoughts?

**Comment by kbarnhart** [comment]
@rjleveque  I've tested this and it works well for me. 

My only thoughts are 

(1) that we should remove the two tests because they are not really of dclaw. Should these not go in the geoclaw/tests/geoclaw_radial_slide and geoclaw/tests/geoclaw_bouss_radial_slide?

(2) that we remove the dclaw/src/2d/dig/fgout_module.f90 file. 

Let me know what you think (also tagging @dlgeorge).

Other than these two issues, I think this is ready to merge.

File: src/2d/dig/Makefile.dclaw (hunk 1 above)
Comment (geoflows) [reply src/2d/dig/Makefile.dclaw:33]: seems best to remove it I guess.

**Comment by dlgeorge** [comment]
@rjleveque @kbarnhart . plots look great, and q_out_vars method seems good to me. PR seems good to me (?).

File: src/2d/dig/Makefile.dclaw (hunk 1 above)
Comment (rjleveque) [reply src/2d/dig/Makefile.dclaw:33]: I think that's fine.  The only difference in it was that it set
```
fg%num_vars(1) = 10 ! DIG different number
```
but this is no longer needed with the geoclaw updates.

**Comment by rjleveque** [comment]
@kbarnhart: regarding moving the examples to geoclaw, I put them here since I was using the SWE version to compare to what I got when running D-Claw with mass fraction 0, so that it was pure water. I don't think it really makes sense to include in GeoClaw since it's not a situation most GeoClaw users would be setting up, although I suppose it could be interesting as a radially symmetric overland flood example.

At any rate I'm fine with removing them from the `tests/` directory since normally we use this directory for unit tests (and hopefully we'll get continuous integration working again soon so these are used more regularly).  Maybe they should be in a new subdirectory `dev/` to keep them around for use in further developing the code?  Or I'm fine with removing them from the repo and I'll just keep them locally.

(The `geoclaw_bouss_radial_slide` was committed by accident since we don't have bouss in D-Claw yet for comparison.)

**Comment by kbarnhart** [comment]
@rjleveque : @dlgeorge and I discussed and we decided that it made sense to keep them here, broadly for the reasons you list. I think keeping geoclaw_bouss_radial_slide makes sense (eventually) when dclaw-bouss exists.

So I went ahead and merged it. Thanks for this PR!

**Pull request merged**